src/App.js

```diff
@@ -1,23 +1,83 @@
-import logo from './logo.svg';
 import './App.css';
+import { useState, useEffect } from 'react';
+import { getDatabase, onValue, ref} from 'firebase/database';
+import { Route, Routes } from 'react-router-dom';
+import firebase from './firebase';
+import ArticleHeadlines from './components/ArticleHeadlines';
+import Header from './components/Header';
+import Footer from './components/Footer';
+import ArticlePage from './components/ArticlePage';
+import Pages from './components/Pages';
+import Filter from './components/Filter';
+
+
+const App = () => {
+  
+  const [articleData, setArticleData] = useState({});
+  const [originalArticleData, setOriginalArticleData] = useState({});
+  const [articleColor, setArticleColor] = useState('#EBEBEB');
+
+  useEffect(() => {
+    const database = getDatabase(firebase);
+    const dbRef = ref(database)
+
+    onValue(dbRef, res => {
+      const data = res.val();
+      const newArticleData = {};
+      for (let key in data) {
+        newArticleData[key] = [];
+        for (let id in data[key]){
+          newArticleData[key].push({...data[key][id], keyId: id});
+        }
+      }
+      newArticleData.articles.sort((a, b) => new Date(b.date) - new Date(a.date));
+      newArticleData.articles = addPropertiesToObject(newArticleData.articles)
+      setOriginalArticleData(newArticleData);
+      setArticleData(newArticleData);     
+    })
+  },[]);
+
+  const filterArticles = (filterBy, filterValue) => {
+    const filteredArticleData = {...originalArticleData};
+    const filteredArticles = filteredArticleData.articles.filter(article => article[filterBy] === filterValue)
+    filteredArticleData.articles = addPropertiesToObject(filteredArticles);
+    setArticleData(filteredArticleData);
+  }
+
+  const addPropertiesToObject = articles => {
+    let i = 0; // amount of spce on page taken by article headline
+    let order = 0;
+    return(
+    articles.map( article => {
+      if (article.bannerSize === 'full') {
+        i = i + 4;
+        i % 4 === 0 ? order = i : order = i - 5;
+      } else {
+        i = i + 2;
+        order = i;
+      }
+      return (
+        { ...article, order: order, articleSlug: `${article.title.toLowerCase().replace(/[^\w\s-]/g, '').replace(/[\s_-]+/g, '-').replace(/^-+|-+$/g, '')}` }
+      )
+    }))
+  }
+
+  const filterForm = () => <Filter resetFilter={resetFilter} filterArticles={filterArticles} articleData={articleData} />
+  const resetFilter = () => setArticleData(originalArticleData);
+  const changeBorderColor = categoryColor => setArticleColor(categoryColor);
+  const pageButtons = pageNum => <Pages articles={articleData.articles} pageNum={pageNum} />;
 
-function App() {
   return (
```

Comment:
I would pseudo-code this out a bit more to indicate what each component is from which js page ie)

return (
//user sees this from ___.js

//then this from ____.js